src/modules/course/course.controller.ts

```diff
@@ -23,12 +24,26 @@ import { CreateReviewDto } from './dto/create-review.dto';
 import { UpdateReviewDto } from './dto/update-review.dto';
 import { courseId } from './docUtils/course.paramdocs';
 import responsedoc from './docUtils/apidoc';
+import { GetCourseFilterDto } from './dto/course-filter.dto';
 
 @ApiTags('Course')
 @Controller('course')
 export class CourseController {
   constructor(private courseService: CourseService) {}
 
+  // get brief information on cards for all courses initially
+  @Get('/cards/all')
+  @ApiOkResponse(responsedoc.getAllCourses)
+  async getBreifAllCourses() {
+    return await this.courseService.getBreifAllCourses();
+  }
+
+  // fetch query results for search string
+  @Get('/all/query')
+  async getSearchResults(@Query() filterDto: GetCourseFilterDto) {
+    return await this.courseService.getSearchResults(filterDto);
+  }
```

Comment:
doc not added

Reply:
Oh
This was the recent one so i think i forgot adding the doc for it.
Will add this too